src/test/java/net/satago/tapestry5/jpa/test/NestedServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package net.satago.tapestry5.jpa.test;
+
+import javax.persistence.EntityManager;
+
+import net.satago.tapestry5.jpa.test.entities.ThingTwo;
+
+import org.apache.tapestry5.jpa.annotations.CommitAfter;
+
+public class NestedServiceImpl implements NestedService {
+	private final EntityManager em;
+
+	public NestedServiceImpl(EntityManager em) {
+		this.em = em;
+	}
+
+	@CommitAfter
```

Comment:
I don't think it has any effect if you put `@CommitAfter` annotation here, because you already have one on the interface.
If I remember correct Tapestry can copy annotations from implementation classes to interfaces since T5.4, but then you don't need the one on the interface?

Same for the `TopLevelService/Impl`, especially for `createThingOneThenTwo` that only has `@CommitAfter` on implementation.